codegen/templates/code/types/json.gen.gotmpl

```diff
@@ -33,7 +36,11 @@ func (this *{{ $resource.Spec.Type.Name }}) MarshalJSON() ([]byte, error) {
 
 // UnmarshalJSON is a custom unmarshaler for {{ $resource.Spec.Type.Name }}
 func (this *{{ $resource.Spec.Type.Name }}) UnmarshalJSON(b []byte) error {
+{{- if $resource.StrictUnmarshal }}
```

Comment:
I like this but do we also feel like adding something in utils or on the codegenned method set to identify the extra fields?

like loose unmarshal, remarshal to json and then compare or some such.

Reply:
I suppose, as mentioned in the comment below, this change has a very specific purpose, so we do not need that. If it becomes necessary I'm happy to help add it.

Reply:
tbh i don't think we have any need for this at all. however if we feel like leaving the option for some unforseen edge case, i think the simplicity here is good enough